Backend/ServiceLayer/Services/ResetService.cs

```diff
@@ -0,0 +1,229 @@
+            List<string> securityQuestions = _resetRepo.GetSecurityQuestions(userIDToGetQuestionsFor);
+            return securityQuestions;
+        }
+
+        //Function to check the security answers of the user associated with the resetID
+        public bool CheckSecurityAnswers(string resetID, List<string> userSubmittedAnswers)
+        {
+            //Get the UserID of the user whose security answers we're checking against
+            Guid userIDToCheckAnswersFor = _resetRepo.GetUserID(resetID);
+            //Check the security answers
+            return _resetRepo.CheckSecurityAnswers(userIDToCheckAnswersFor, userSubmittedAnswers);
+        }
+
+        //Function to lock the resetID for 24 hours if failed to answer security questions 3 times
+        public void LockResetID(string resetID)
+        {
+            _resetRepo.LockOut(resetID);
+        }
+
+        //Function to check if the resetID is locked out
+        public bool IsLockedOut(string resetID)
+        {
+            return _resetRepo.CheckLockOut(resetID);
+        }
+
+        //Function to check how many times a reset has been attempted with the resetID
+        public int GetResetCount(string resetID)
+        {
+            return _resetRepo.GetAttemptsPerResetID(resetID);
+        }
+
+        //Function to see how many resetIDs are associated with the given email
+        public int GetResetIDCount(string email)
+        {
+            return _resetRepo.GetResetIDCountPerEmail(email);
+        }
+        
+        #region Email Functions
+        //Function to create the email is user exists 
+        public void SendResetEmailUserExists(string receiverEmail, string resetURL)
+        {
+            //Need SQL Query to get info about user from DB
+            string userFirstName = "";
+            string userlastName = "";
+            string userFullName = userFirstName + " " + userlastName;
+            string template = "Hi {0}, \r\n" +
+                                             "You recently requested to reset your password for your KFC account, click the link below to reset it.\r\n" +
+                                             "The URL is only valid for the next 5 minutes\r\n {1}" +
+                                             "If you did not request to reset your password, please contact us by responding to this email.\r\n\r\n" +
+                                             "Thanks, KFC Team";
+            string data = "userFirstName, resetURL";
+            string resetPasswordBodyString = string.Format(template, data);
+            //Create the email service object
+            EmailService es = new EmailService();
+            //Create the message that will be sent
+            MimeMessage emailToSend = es.createEmailPlainBody(userFullName, receiverEmail, resetPasswordSubjectString, resetPasswordBodyString);
+            //Send the email with the message
+            es.sendEmail(emailToSend);
```

Comment:
As of now, looks like this is only sending a plaintext email - see my comment above in the email service.